enhancements/ingress/haproxy-healthcheckinterval.md

```diff
@@ -0,0 +1,291 @@
+identified as a cause for concern for all customers, chiefly because we set it to 5 seconds.  The HAProxy default
+is 2 seconds, and an interval of 10 seconds is documented in
+[HAProxy configuration manual](http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#3.2-max-spread-checks)
+as a **large** check interval.
+
+Performance validation is not in scope for this proposal.
+
+Other parameters may come into play for adjusting the healthcheck interval, but will not be covered in this proposal.
+It is worth checking in the future, for the impact of `spread-checks`, `max-spread-checks`, `fastinter`, `downinter`,
+and the use of passive health checks rather than active health checks.
+
+## Proposal
+
+HAProxy configures health check intervals using the
+[`check` parameter](http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#5.2-check).
+- `check inter <duration>`: sets the interval between two consecutive health checks to the time specified
+in `duration`.
+
+For plain HTTP backends, or backends with TLS termination set to edge, re-encrypt, or passthrough,
+the `check inter` duration value may come from (evaluated in order):
+- the `router.openshift.io/haproxy.health.check.interval` annotation on the route
+- a (currently hidden) `ROUTER_BACKEND_CHECK_INTERVAL` environment variable
+- 5 seconds as a default if the previous two values are unset
+
+Note that this `check inter` value also serves as a timeout for health checks sent to servers
+if `timeout check` is not set, but `timeout check` is currently hard-coded to 5 seconds in the OpenShift HAProxy
+configuration.  The minimum value for `check inter` will be 5 seconds.
+(See the Risks and Mitigations section for further discussion.)
+
+`check inter` is set per server for the same backends as `timeout check`, with some
+additional constraints.  For plain HTTP backends, or backends with TLS termination set to edge, re-encrypt, or
+passthrough, it is set only if the endpoint is not idled, and the route has more than one active endpoint.
+
+This proposal will add a new optional parameter, `HealthCheckInterval` to the `IngressControllerTuningOptions` struct,
+to set the duration used in `check inter` at the backend scope for every backend server that currently sets a healthcheck
+interval.
+
+### User Stories
+
+#### As a cluster administrator, I need to reduce the frequency of healthchecks to 20 seconds on all backends
+
+Edit the `IngressController` specification to add `healthCheckInterval` to `tuningOptions`.  The duration of the
+`healthCheckInterval` must be set as a string representing time values.  The time value format is an integer optionally
+followed by a time unit (e.g. "ms", "s", "m", etc.). If no unit is specified, the value is measured in
+milliseconds. More information on the time format can be found in the
+[HAProxy documentation](https://github.com/haproxy/haproxy/blob/v2.2.0/doc/configuration.txt).
```

Comment:
The API uses the `metav1.Duration` value format.  For a citation, we could use <https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Duration> and <https://pkg.go.dev/time#ParseDuration>.  The most important difference from the HAProxy format is that the unit is required.